File: scan.dart
#!/usr/bin/env dart

import 'dart:async';
import 'dart:io';

const host = "192.168.0.200";
const timeout = Duration(milliseconds: 750);

Future check(port) async {
  return Socket.connect(host, port, timeout: timeout).then((s) {
    print(s.remotePort);
    s.destroy();
  }).catchError((e) => null);
}

Future main() async {
  var ports = new List<int>.generate(128, (i) => i + 1);
  return Future.wait(ports.map(check));
}

File: scan.mjs
// true 2>/dev/null; exec /usr/bin/env node "$0" "$@"

import { Socket } from 'net'

const HOST = '192.168.0.200';

async function check(host, port) {
    let socket = new Socket().setTimeout(750);

    socket.on('timeout', socket.destroy);
    socket.on('error', socket.destroy);

    socket.connect(port, host, function() {
        this.destroy();
        console.log(port);
    });
};

const tasks = [...Array(128)].map((_, i) => check(HOST, i + 1));

for await (const _ of tasks);
